### PR TITLE
One fix per day 18

### PR DESCRIPTION
**Pull request by mbaechler**
only boring refactoring here, no scala, no implicit, nothing shiny

### CODE REVIEW

**Comment by mbaechler** [comment]
many tests failing in `apache-james-mailbox-elasticsearch` for good reasons. Will handle later.

File: mailet/base/src/main/java/org/apache/mailet/base/GenericMailet.java [hunk 1]
```diff
@@ -290,34 +290,18 @@ public void log(String message, Throwable t) {
...
-        while (iterator.hasNext()) {
-            String parameter = iterator.next();
-            if (!allowed.contains(parameter)) {
-                bad.add(parameter);
-            }
-        }
-        
-        if (bad.size() > 0) {
-            throw new MessagingException("Unexpected init parameters found: " + Arrays.toString(bad.toArray()));
+        if (!bad.isEmpty()) {
```
Comment (nvduc91) [inline mailet/base/src/main/java/org/apache/mailet/base/GenericMailet.java:302]: I think `bad` is not a good name, how about `reject` or something more clear?

Comment (mbaechler) [reply mailet/base/src/main/java/org/apache/mailet/base/GenericMailet.java:302]: I didn't took care of that originally and to be honest, refactoring is an activity requiring to not care about everything if one want to settle to a PR.
I propose to not change that in this PR and you create a PR with this change if you care enough about it.

Comment (nvduc91) [reply mailet/base/src/main/java/org/apache/mailet/base/GenericMailet.java:302]: ok.

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**